src/test/groovy/org/prebid/server/functional/tests/privacy/PrivacyOrtbConverterSpec.groovy

```diff
@@ -0,0 +1,291 @@
+        def prepareEncodeResponseBodyWithLegIntAndFlexiblePurposes = getVendorListContent(false, true, true)
+        def prepareEncodeResponseBodyWithPurposesAndFlexiblePurposes = getVendorListContent(true, false, true)
+        privacyPbsContainerWithMultipleGvlWithElderOrtb = new PrebidServerContainer(PBS_CONFIG + ["adapters.generic.ortb-version": "2.5"])
+        privacyPbsContainerWithMultipleGvlWithElderOrtb.withCopyToContainer(Transferable.of(prepareEncodeResponseBodyWithPurposesOnly), getVendorListPath(PURPOSES_ONLY_GVL_VERSION))
+        privacyPbsContainerWithMultipleGvlWithElderOrtb.withCopyToContainer(Transferable.of(prepareEncodeResponseBodyWithLegIntPurposes), getVendorListPath(LEG_INT_PURPOSES_ONLY_GVL_VERSION))
+        privacyPbsContainerWithMultipleGvlWithElderOrtb.withCopyToContainer(Transferable.of(prepareEncodeResponseBodyWithLegIntAndFlexiblePurposes), getVendorListPath(LEG_INT_AND_FLEXIBLE_PURPOSES_GVL_VERSION))
+        privacyPbsContainerWithMultipleGvlWithElderOrtb.withCopyToContainer(Transferable.of(prepareEncodeResponseBodyWithPurposesAndFlexiblePurposes), getVendorListPath(PURPOSES_AND_LEG_INT_PURPOSES_GVL_VERSION))
+        privacyPbsContainerWithMultipleGvlWithElderOrtb.start()
+        privacyPbsServiceWithMultipleGvlWithElderOrtb = new PrebidServerService(privacyPbsContainerWithMultipleGvlWithElderOrtb)
+    }
+
+    def cleanupSpec() {
+        privacyPbsContainerWithMultipleGvlWithElderOrtb.stop()
+    }
+
+    def "PBS should leave the original request with ext.eids data for elder ortb when requireConsent is enabled and #enforcementRequirements.purpose have any basic consent"() {
+        given: "Default Generic BidRequests with Eid field"
+        def userEids = [Eid.defaultEid]
+        def tcfConsent = TcfUtils.getConsentString(enforcementRequirements)
+        def bidRequest = getGdprBidRequest(tcfConsent).tap {
+            it.ext.prebid.trace = VERBOSE
+            it.user.eids = userEids
+        }
+
+        and: "Save account config with requireConsent into DB"
+        def purposes = TcfUtils.getPurposeConfigsForPersonalizedAds(enforcementRequirements, true)
+        def accountGdprConfig = new AccountGdprConfig(purposes: purposes, basicEnforcementVendors: [GENERIC.value])
+        def activity = Activity.getDefaultActivity([ActivityRule.getDefaultActivityRule(Condition.baseCondition, true)])
+        def account = getAccountWithGdpr(bidRequest.accountId, accountGdprConfig).tap {
+            config.privacy.allowActivities = AllowActivities.getDefaultAllowActivities(TRANSMIT_EIDS, activity)
+        }
+        accountDao.save(account)
+
+        when: "PBS processes auction requests"
+        activityPbsServiceExcludeGvlWithElderOrtb.sendAuctionRequest(bidRequest)
+
+        then: "Generic bidder request should have data in Eid field"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest?.user?.ext?.eids == userEids
```

Comment:
Can you add an assertion for `bidderRequest.user.eids`?
Check other occurrences.